specs/bytecode-proof.md

```diff
@@ -0,0 +1,57 @@
+# Bytecode Proof
+
+The bytecode proof helps the EVM proof by making the bytecode (identified by its code hash) easily accessible. Each byte in the bytecode is made accessible with its position in the bytecode. It is also annotated with a flag indicating if the byte is an opcode or is simply data for a previous PUSH instruction.
+
+## Circuit Layout
+
+| Column                | Description                                                        |
+| --------------------- | ------------------------------------------------------------------ |
+| `q_first`             | `1` on the first row, else `0`                                     |
+| `q_last`              | `1` on the last row, else `0`                                      |
+| `hash`                | The keccak hash of the bytecode                                    |
+| `index`               | The position of the byte in the bytecode                           |
+| `byte`                | The byte data for the current position                             |
+| `is_code`             | `1` if the byte is code, `0` if the byte is PUSH data              |
+| `push_data_left`      | The number of bytes that still need to follow for PUSH data        |
+| `hash_rlc`            | The accumulator containg the current and previous bytes            |
+| `hash_length`         | The bytecode length                                                |
+| `byte_push_size`      | The number of bytes pushed for the current byte                    |
+| `is_final`            | `1` if the current byte is the last byte of the bytecode, else `0` |
+| `padding`             | `1` if the current row is padding, else `0`                        |
```

Comment:
Perhaps a couple of words why padding is needed, it's because lookup into keccak_table fails for zero rows, right?

Reply:
Yeah didn't want to make assumptions, added this: 
> Explicit padding is added to this circuit to be able to fully fill the circuit with valid data without depending on the keccak circuit to either hash this padding data or support looking up e.g. all zero data.

Reply:
Cool, thanks!